code/game/machinery/suit_storage_unit.dm

```diff
@@ -451,11 +451,9 @@
 		if(mob_occupant)
 			things_to_clear += mob_occupant
 			things_to_clear += mob_occupant.GetAllContents()
-		for(var/atom/movable/AM in things_to_clear) //Scorches away blood and forensic evidence, although the SSU itself is unaffected
-			SEND_SIGNAL(AM, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRONG)
-			var/datum/component/radioactive/contamination = AM.GetComponent(/datum/component/radioactive)
-			if(contamination)
-				qdel(contamination)
+		for(var/am in things_to_clear) //Scorches away blood and forensic evidence, although the SSU itself is unaffected
+			var/atom/movable/dirty_movable = am
+			dirty_movable.wash(CLEAN_ALL)
```

Comment:
One thing I don't like about this thing: it has a half-exploit where you can shove someone in here and decontaminate to instantly remove all radiation from them.

Reply:
Hm? I thought that was sort of the purpose of the decontamination mode on these? 

It decontaminates and clears all grime, including radioactive particles from whatever is inside, with it burning the shit out of someone if theres an occupant.